src/bootstrap/lib.rs

```diff
@@ -607,8 +607,15 @@ impl Build {
 
     /// Get the space-separated set of activated features for the standard
     /// library.
-    fn std_features(&self) -> String {
+    fn std_features(&self, compiler: &Compiler) -> String {
         let mut features = "panic-unwind".to_string();
+
+        // The sanitizer runtimes depend on LLVM, which gets build during stage0
+        // rustc
+        if compiler.stage != 0 {
+            features.push_str(" asan lsan msan tsan")
+        }
```

Comment:
We could set these features for all stages but do nothing (generate an empty rlib) during stage0.